### PR TITLE
Added search functionality

### PR DESCRIPTION
**Pull request by filipemir**

### CODE REVIEW

File: app/controllers/elevators_controller.rb [hunk 1]
```diff
@@ -50,6 +50,16 @@ def update
     end
   end
 
+  def search
+    @elevators = Elevator.find_by_fuzzy_building_name(
+      params['search'].to_s
+    )
```
Comment (kevlarr) [inline app/controllers/elevators_controller.rb]: This is only a single argument, and the line's not too long, so it looks a little weird as a multiliner... 
```
@elevators = Elevator.find_by_fuzzy_building_name(params['search'].to_s)
```
... or...
```
@elevators =
  Elevator.find_by_fuzzy_building_name(params['search'].to_s)
```
.. would be preferable here.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -8,22 +8,33 @@
...
+      <%= submit_tag "Search" %>
+
+    <% end %>
+  </div>
+
+  <div class="nav-buttons">
+    <% if current_admin %>
+      <%= link_to "Admins", admins_path %>
+    <% end %>
+    <% if current_user || current_admin %>
```
Comment (kevlarr) [inline app/views/layouts/application.html.erb]: Is `|| current_admin` necessary here? Won't `current_user` be present if an admin's logged in?

Comment (filipemir) [inline app/views/layouts/application.html.erb]: Not in the way we have it set up. Admins and users are separate models. So they're mostly distinct. You could actually have a user and an admin signed in at the same time with our schema. This way we could separate the functionality for each.

File: app/controllers/elevators_controller.rb (hunk 1 above)
Comment (filipemir) [inline app/controllers/elevators_controller.rb]: Makes sense. Fixed!

File: app/models/trigram.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+class Trigram < ActiveRecord::Base
+  include Fuzzily::Model
+end
```
Comment (filipemir) [inline app/models/trigram.rb]: No it's not

File: spec/features/search_spec.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    click_on 'Stinky yet NICE elevator'
+    expect(page).to have_content('Stinky yet NICE elevator')
+    expect(page).to_not have_content('Nice building with nice elevator')
+    expect(page).to have_content('teststate')
+    expect(page).to have_content('testcity')
+    expect(page).to have_content('Address: testaddress')
+  end
+
+end
```
Comment (filipemir) [inline spec/features/search_spec.rb]: No it's not

File: spec/features/search_spec.rb [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    FactoryGirl.create(:elevator, building_name: 'Stinky elevator')
+    FactoryGirl.create(:elevator, building_name: 'Stinky yet NICE elevator')
+    FactoryGirl.create(:elevator, building_name: 'Idiotic building')
+    FactoryGirl.create(:elevator, building_name: 'The White House')
+    FactoryGirl.create(:elevator, building_name: 'The White Mountains')
+    FactoryGirl.create(:elevator, building_name: 'The White Whale')
+  end
+
+  scenario 'Non-logged-in user searches for a specific elevator' do
+    visit '/'
```
Comment (justinhuynh) [inline spec/features/search_spec.rb]: change `'/'` to path helper (`root_path`)

File: spec/features/search_spec.rb [hunk 6]
```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+
+feature 'Search bar:' do
+
+  before :each do
+    FactoryGirl.create(:user, email: 'email@email.com', password: 'password')
```
Comment (justinhuynh) [inline spec/features/search_spec.rb]: This is something that can definitely get refactored later - you could have an array of names and then have a loop in your factory that creates these items...we can discuss next time

Comment (filipemir) [inline spec/features/search_spec.rb]: Good call. I made this change

File: spec/features/search_spec.rb (hunk 5 above)
Comment (filipemir) [inline spec/features/search_spec.rb]: done

File: spec/features/search_spec.rb [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    click_on 'Search'
+
+    expect(page).to have_css('.elevator', count: 3)
+    expect(page).to have_content('The White House')
+    expect(page).to have_content('The White Mountains')
+    expect(page).to have_content('The White Whale')
+  end
+
+  scenario 'Admin searches for a specific elevator' do
+    visit '/admins/sign_in'
```
Comment (justinhuynh) [inline spec/features/search_spec.rb]: change this to path helper (next time)

**Pull request merged**